src/test/spec/retryable_writes/mod.rs

```diff
@@ -0,0 +1,316 @@
+            vec![
+                doc! { "$match": doc! { "x": 1 } },
+                doc! { "$merge": "other_coll" },
+            ],
+            None,
+        )
+        .await
+        .unwrap();
+        assert!(excludes("aggregate"));
+    }
+}
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+#[function_name::named]
+async fn transaction_ids_included() {
+    let _guard = LOCK.run_concurrently().await;
+
+    let client = EventClient::new().await;
+
+    if !(client.is_replica_set() || client.is_sharded()) {
+        return;
+    }
+
+    let coll = client.init_db_and_coll(function_name!(), "coll").await;
+
+    let includes = |command_name: &str| -> bool {
+        let (started, _) = client.get_successful_command_execution(command_name);
+        started.command.contains_key("txnNumber")
+    };
+
+    coll.insert_one(doc! { "x": 1 }, None).await.unwrap();
+    assert!(includes("insert"));
+
+    coll.update_one(doc! {}, doc! { "$set": doc! { "x": 1 } }, None)
+        .await
+        .unwrap();
+    assert!(includes("update"));
+
+    coll.replace_one(doc! {}, doc! { "x": 1 }, None)
+        .await
+        .unwrap();
+    assert!(includes("update"));
+
+    coll.delete_one(doc! {}, None).await.unwrap();
+    assert!(includes("delete"));
+
+    coll.find_one_and_delete(doc! {}, None).await.unwrap();
+    assert!(includes("findAndModify"));
+
+    coll.find_one_and_replace(doc! {}, doc! { "x": 1 }, None)
+        .await
+        .unwrap();
+    assert!(includes("findAndModify"));
+
+    coll.find_one_and_update(doc! {}, doc! { "$set": doc! { "x": 1 } }, None)
+        .await
+        .unwrap();
+    assert!(includes("findAndModify"));
+
+    let options = InsertManyOptions::builder().ordered(true).build();
+    coll.insert_many(vec![doc! { "x": 1 }], options)
+        .await
+        .unwrap();
+    assert!(includes("insert"));
+
+    let options = InsertManyOptions::builder().ordered(false).build();
+    coll.insert_many(vec![doc! { "x": 1 }], options)
+        .await
+        .unwrap();
+    assert!(includes("insert"));
+}
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+#[function_name::named]
+async fn mmapv1_error_raised() {
+    let _guard = LOCK.run_concurrently().await;
+
+    let client = TestClient::new().await;
+
+    let req = semver::VersionReq::parse("<=4.0").unwrap();
+    if !req.matches(&client.server_version) || !client.is_replica_set() {
+        return;
+    }
+
+    let coll = client.init_db_and_coll(function_name!(), "coll").await;
+
+    let server_status = client
+        .database(function_name!())
+        .run_command(doc! { "serverStatus": 1 }, None)
+        .await
+        .unwrap();
+    let name = server_status
+        .get_document("storageEngine")
+        .unwrap()
+        .get_str("name")
+        .unwrap();
+    if name != "mmapv1" {
+        return;
```

Comment:
AFAIK this test is always returning here on Evergreen since wiredTiger is being used, but the test passes locally when `--storagEngine=mmapv1` is passed to mongod. Is there anything we should do to test this in CI?

Reply:
We could add an mmapv1 task, but honestly it's probably not worth the effort (and evergreen cost). If a user ends up reporting a bug with mmapv1, we can revisit, but I don't expect that we have many mmapv1 users (if any at all).